src/app/GitCommands/RevisionReader.cs

```diff
@@ -608,45 +608,17 @@ int CountParents(in ReadOnlySpan<byte> array, int baseOffset)
                 currentOffset++;
             }
 
-            // Removes empty Notes markers (this is the most common case)
-            bool hasNonEmptyNotes = _hasNotes;
-            if (hasNonEmptyNotes)
+            if (_hasNotes)
             {
-                if (decoded.EndsWith(_notesMarker))
+                if (!decoded.EndsWith(_notesMarker) && ((ReadOnlySpan<char>)decoded).LastIndexOf(_notesMarker, StringComparison.Ordinal) is int notesStartIndex and >= 0)
                 {
-                    // Remove the empty marker
-                    decoded = decoded[..^_notesMarker.Length].TrimEnd();
-                    hasNonEmptyNotes = false;
+                    revision.Body = decoded[..notesStartIndex].TrimEnd().ToString();
+                    revision.Notes = decoded.Slice(notesStartIndex + _notesMarker.Length + 1).ToString();
```

Comment:
The first newline was not removed
```suggestion
                    revision.Notes = decoded.Slice(1 + notesStartIndex + _notesMarker.Length + 1).ToString();
```